agent/stats/container.go

```diff
@@ -107,6 +108,34 @@ func (container *StatsContainer) collect() {
 	}
 }
 
+func (container *StatsContainer) getApiContainer(dockerID string) (*apicontainer.Container, error) {
+	if container.resolver == nil {
+		return nil, nil
+	}
+	task, err := container.resolver.ResolveTask(dockerID)
```

Comment:
If I understand correctly, using `container.resolver.ResolveTask` and then iterating through all the containers of the task seems unnecessary. Could we leverage using `container.resolver.ResolveContainer` instead (directly if possible), which should also result in a more desirable time complexity?

Reply:
Unfortunately ResolveContainer returns a DockerContainer object rather than an apicontainer.Container object, which is what we need here.

Reply:
Doesn't `DockerContainer` itself contain the `Container` type that is needed? Could we make use of this or is it not possible?

https://github.com/aws/amazon-ecs-agent/blob/dev/agent/api/container/container.go#L353

Reply:
yes you are right, I will change this